src/functions.jl

```diff
@@ -289,20 +289,62 @@ function sum_dict(kvs::Vector{Pair{K, V}}) where {K, V}
     d
 end
 
-# _pair transforms a *Term into a pair key => coefficient where the key groups the variables of the term and its output_index if it is not one.
-function _pair(t::Union{VectorAffineTerm, VectorQuadraticTerm})
-    p = _pair(t.scalar_term)
-    (t.output_index, p.first) => p.second
+"""
+    coefficient(t::Union{ScalarAffineTerm, ScalarQuadraticTerm
+                         VectorAffineTerm, VectorQuadraticTerm})
+
+Finds the coefficient associated with the term `t`.
```

Comment:
"associated with" implies lookup in a container. It's clearer to say "stored in".